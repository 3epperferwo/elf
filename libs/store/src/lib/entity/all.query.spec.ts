import { createEntitieStore, createTodo } from '../mocks/stores.mock';
import { addEntity } from './add.mutation';
import { selectAll } from './all.query';

describe('selectAll', () => {
  let store: ReturnType<typeof createEntitieStore>;

  beforeEach(() => {
    store = createEntitieStore();
  });

  it('should query all', () => {
    store.pipe(selectAll()).subscribe(value => {
      expect(value).toMatchSnapshot();
    });

    store.reduce(
      addEntity(createTodo(2))
    );
  });

});
