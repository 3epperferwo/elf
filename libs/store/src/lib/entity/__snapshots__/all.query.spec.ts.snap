// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectAll should query all 1`] = `Array []`;

exports[`selectAll should query all 2`] = `
Array [
  Object {
    "completed": false,
    "id": 2,
    "title": "todo 2",
  },
]
`;
